packages/flutter/lib/src/material/data_table.dart

```diff
@@ -1036,6 +1080,7 @@ class DataTable extends StatelessWidget {
         sorted: dataColumnIndex == sortColumnIndex,
         ascending: sortAscending,
         overlayColor: effectiveHeadingRowColor,
+        headingLabel: column.heading != null ? column.heading! : '',
```

Comment:
it looks like this is only for semantics announcement? The name should have `semantics` in it. `label` also imply this is semantics.label which may not be the case?